api/app/db/crud/hfi_calc.py

```diff
@@ -77,14 +79,49 @@ def get_most_recent_updated_hfi_request_for_current_date(session: Session,
 
 def store_hfi_request(session: Session, hfi_result_request: HFIResultRequest, username: str):
     """ Store the supplied hfi request """
-    hfi_request = HFIRequest(
-        fire_centre_id=hfi_result_request.selected_fire_center_id,
-        prep_start_day=hfi_result_request.date_range.start_date,
-        prep_end_day=hfi_result_request.date_range.end_date,
-        create_timestamp=get_utc_now(),
-        create_user=username,
-        request=hfi_result_request.json())
-    session.add(hfi_request)
+    stmt = (
+        insert(HFIRequest).values(fire_centre_id=hfi_result_request.selected_fire_center_id,
+                                  prep_start_day=hfi_result_request.date_range.start_date,
+                                  prep_end_day=hfi_result_request.date_range.end_date,
+                                  create_timestamp=get_utc_now(),
+                                  create_user=username,
+                                  request=hfi_result_request.json()).returning(HFIRequest.id)
+    )
+    res = session.execute(stmt).fetchone()
+    latest_hfi_request_id = res.id
+    latest_hfi_ready_records = get_latest_hfi_ready_records(session, hfi_result_request)
+    if latest_hfi_ready_records is not None:
+        updated_hfi_ready_records = []
+        for hfi_ready_record in latest_hfi_ready_records:
+            updated_hfi_ready_records.append(HFIReady(hfi_request_id=latest_hfi_request_id,
+                                                      planning_area_id=hfi_ready_record.planning_area_id,
+                                                      ready=hfi_ready_record.ready,
+                                                      create_timestamp=hfi_ready_record.create_timestamp,
+                                                      create_user=hfi_ready_record.create_user,
+                                                      update_timestamp=hfi_ready_record.update_timestamp,
+                                                      update_user=hfi_ready_record.update_user))
+        session.bulk_save_objects(updated_hfi_ready_records)
+
+
+def get_latest_hfi_ready_records(session: Session, hfi_result_request: HFIResultRequest):
+    """ Retrieve the current hfi ready records for each planning area based latest
+        HFI request defined by fire centre and prep date range. 
+    """
+    query = session.query(HFIReady)\
+        .join(HFIRequest)\
+        .filter(HFIRequest.fire_centre_id == hfi_result_request.selected_fire_center_id)\
+        .filter(HFIRequest.prep_start_day == hfi_result_request.date_range.start_date)\
+        .filter(HFIRequest.prep_end_day == hfi_result_request.date_range.end_date)\
+        .distinct(HFIRequest.fire_centre_id,
```

Comment:
You don't need a distinct here - I would either modify get_most_recent_updated_hfi_request to return the request and the ready - or, I think better, use the id of the most recent hfi_result_request and just filter on it.